api4/plugin.go

```diff
@@ -81,6 +83,51 @@ func uploadPlugin(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(manifest.ToJson()))
 }
 
+func installPluginFromUrl(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !*c.App.Config().PluginSettings.Enable {
+		c.Err = model.NewAppError("installPluginFromUrl", "app.plugin.disabled.app_error", nil, "", http.StatusNotImplemented)
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(c.App.Session, model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+
+	url := c.Params.PluginDownloadUrl
+
+	if !model.IsValidHttpUrl(url) {
+		c.Err = model.NewAppError("installPluginFromUrl", "api.plugin.install.invalid_url.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	if !(strings.HasPrefix(url, "https") || *c.App.Config().PluginSettings.AllowInsecureDownloadUrl) {
```

Comment:
Maybe use https://golang.org/pkg/net/url and compare the scheme instead of checking a prefix? Both will likely work but using the URL package seems cleaner.